src/bun.js/node/node_fs.zig

```diff
@@ -5094,6 +5084,38 @@ pub const NodeFS = struct {
         };
     }
 
+    /// Calculate Base64 encoded string from an ArrayList of bytes.
+    /// Conversion happens in place, in the same input array, with an hardcoded chunk size of 16383 bytes.
+    /// Memory is allocated as needed and not at once time, preventing double allocations for input and output.
+    /// Returns an owned slice of the converted input.
+    fn base64Conversion(buf: *std.ArrayList(u8)) anyerror![]u8 {
+        // Chunk size should be a multiple of 3 bytes to be able to split base64 calculation
+        const chunk_size = 16383;
+        const encoded_chunk_size = comptime Base64.encodeLenFromSize(chunk_size);
+        var offset: usize = 0;
+
+        // Declare buffer to store encoded Base64 Chunk over every chunked input
+        const temp_encoding_buffer = bun.default_allocator.alloc(u8, encoded_chunk_size) catch |err| return err;
+        defer bun.default_allocator.free(temp_encoding_buffer);
+
+        while (true) {
+            const remaining_bytes = buf.items.len - offset;
+            const chunk_to_be_encoded = if (remaining_bytes <= chunk_size) {
```

Comment:
the `if` expression here is confusing at first to read because one branch of the if is `noreturn` while the other is `[]const u8`. can you rewrite this to just the if statement with the early return followed by `const chunk_to_be_encoded = buf.items[offset..(offset + chunk_size)];`

Reply:
Agree, extracted the if statement